Controller/Callback/Offsite.php

```diff
@@ -60,6 +61,10 @@ public function execute()
     {
         $order = $this->session->getLastRealOrder();
 
+        if ($order->getState() === Order::STATE_PROCESSING) {
```

Comment:
@jacstn I'm still not sure if this is a good idea to redirect `Order::STATE_PROCESSING` order to the `PATH_SUCCESS` whenever user enters this callback url.

Just wonder for only one question,
- Will any action from Magento be triggered when user enters the page? (i.e. is there any script that would be executed when the page is opened?)

If nothing at all, Magento just simply renders a HTML success page, then maybe ok... (still not sure, but well, no question further).

Reply:
Also, this condition should be checked after the below condition.
```php
if (! $order->getId()) {
    $this->messageManager->addErrorMessage(__('The order session no longer exists, please make an order again or contact our support if you have any questions.'));
	
    return $this->redirect(self::PATH_CART);
}
```

Reply:
yes.. @guzzilar this callback when successfully done change state of order to `processing` than when already `processing`, no need to do anything.

this action only redirects to success page.

Reply:
Any news on the fix to prevent this 'double redirect' even happening? Feels like we are addressing a symptom rather than the cause

Reply:
backend team did not find any causes yet @jonrandy .